tests/test_result/test_result_methods.py

```diff
@@ -0,0 +1,25 @@
+
+import pytest
+
+from returns.io import IO, IOResult
+from returns.methods import partition
+from returns.result import Failure, Success
+
+
+@pytest.mark.parametrize(('containers', 'expected'), [
+    (
+        (Success(1), Success(2), Failure(None)),
+        ([1, 2], [None]),
+    ),
+    (
+        (
+            IOResult.from_value(1),
+            IOResult.from_value(2),
+            IOResult.from_failure(None),
+        ),
+        ([IO(1), IO(2)], [IO(None)]),
+    ),
+])
+def test_partition(containers, expected):
```

Comment:
Please, also test `Maybe`, since we now work with `Unwrappable`

Reply:
ok, but I can't make Nothing work

Reply:
Why?

Reply:
can't create type for maybe test that is ok for mypy 
Neither
`x: Tuple[Some[int], Some[int], Maybe[None]]`
nor 
`x: Tuple[Some[int], Some[int], Nothing]`
nor 
`x: Tuple[Some[int], Some[int], _Nothing]`
nor
`x: Tuple[Some[int], Some[int], Maybe[Never]]`

Reply:
Ok, I see. This is a big problem with `tuple`. Use `list` for now. I will think about how we can fix this for `tuple`. But, it will be very hard. I will do that